main.inc.php

```diff
@@ -0,0 +1,293 @@
+		} else {
+			$forum_time_zone = $_SESSION[$settings['session_prefix'].'usersettings']['time_zone'];
+		}
+	} elseif($settings['time_zone'] != '') {
+		date_default_timezone_set($settings['time_zone']);
+		$forum_time_zone = $settings['time_zone'];
+	} else {
+		date_default_timezone_set('UTC');
+		$forum_time_zone = 'UTC';
+	}
+}
+
+// do daily actions:
+//daily_actions(TIMESTAMP);
+
+$categories = get_categories();
+$category_ids = get_category_ids($categories);
+if ($category_ids != false) $category_ids_query = implode(', ', $category_ids);
+if (empty($category)) $category = 0;
+
+// user settings:
+if (isset($_COOKIE[$settings['session_prefix'].'usersettings'])) {
+	$usersettings_cookie = explode('.', $_COOKIE[$settings['session_prefix'].'usersettings']);
+}
+
+if (empty($_SESSION[$settings['session_prefix'].'usersettings'])) {
+	if (isset($usersettings_cookie[0])) {
+		$usersettings['user_view'] = $usersettings_cookie[0] == 1 ? 1 : 0;
+	} else {
+		$usersettings['user_view'] = $settings['default_view'];
+	}
+	$usersettings['thread_order'] = isset($usersettings_cookie[1]) && $usersettings_cookie[1] == 1 ? 1 : 0;
+	$usersettings['sidebar'] = isset($usersettings_cookie[2]) && $usersettings_cookie[2] == 0 ? 0 : 1;
+	if(isset($usersettings_cookie[3])) {
+    $usersettings['fold_threads'] = $usersettings_cookie[3] == 1 ? 1 : 0;
+	} else {
+		$usersettings['fold_threads'] = $settings['fold_threads'];
+	}
+	$usersettings['thread_display'] = isset($usersettings_cookie[4]) && $usersettings_cookie[4] == 1 ? 1 : 0;
+	$usersettings['page'] = 1;
+	$usersettings['category'] = 0;
+	$_SESSION[$settings['session_prefix'].'usersettings'] = $usersettings;
+	setcookie($settings['session_prefix'].'usersettings', $_SESSION[$settings['session_prefix'].'usersettings']['user_view'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_order'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['sidebar'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['fold_threads'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_display'], TIMESTAMP + (3600 * 24 * $settings['cookie_validity_days']));
+}
+
+if (isset($_REQUEST['toggle_sidebar'])) {
+	if (empty($_SESSION[$settings['session_prefix'].'usersettings']['sidebar'])) $_SESSION[$settings['session_prefix'].'usersettings']['sidebar'] = 1;
+	else $_SESSION[$settings['session_prefix'].'usersettings']['sidebar'] = 0;
+	setcookie($settings['session_prefix'].'usersettings', $_SESSION[$settings['session_prefix'].'usersettings']['user_view'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_order'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['sidebar'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['fold_threads'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_display'], TIMESTAMP + (3600 * 24 * $settings['cookie_validity_days']));
```

Comment:
cookie_options - Missing

`setcookie($settings['session_prefix'].'usersettings', $_SESSION[$settings['session_prefix'].'usersettings']['user_view'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_order'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['sidebar'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['fold_threads'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_display'], cookie_options(TIMESTAMP + (3600 * 24 * $settings['cookie_validity_days'])));`

Reply:
It looks to me, like every occurence of missing `cookie_options` is solved. Can we mark *this part* as resolved?

Reply:
Yes, I think so. Go ahead.